src/test/java/org/openhds/service/impl/census/IndividualServiceTest.java

```diff
@@ -34,4 +39,38 @@ protected Individual makeValidEntity(String name, String id) {
     protected void initialize(IndividualService service) {
         this.service = service;
     }
+
+    @Test
+    public void recordWithExistingReferences() {
+
+        //Grab a valid entity
+        Individual individual = makeValidEntity("validName", "validId");
+
+        FieldWorker fieldWorker = individual.getCollectedBy();
+        individual.setCollectedBy(null);
+
+        // pass it all into the record method
+        individual = service.recordIndividual(individual, fieldWorker.getUuid());
+
+
+        //Check that the originals match the ones pulled out from findOrMakePlaceholder()
+        assertNotNull(individual.getCollectedBy());
+        assertEquals(individual.getCollectedBy(), fieldWorker);
+
+    }
+
+    @Test
+    public void recordWithNonexistentReferences(){
+
+        //Make a new entity with no references
+        Individual individual = makeValidEntity("validName", "validId");
+        individual.setCollectedBy(null);
+
+        //Pass it in with new reference uuids
+        individual = service.recordIndividual(individual, "feldwarker");
+
+        //check that they were persisted
+        assertNotNull(individual.getCollectedBy());
```

Comment:
The idea is that "feldwarker" should have been created as a placeholder, right?  I think we should tests that individual.getCollectedBy() does indeed have the id "feldwarker".

This will also make the expected behavior easier to deduce from the test code.